core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/CustomHostNameVerifier.java

```diff
@@ -0,0 +1,28 @@
+package org.wso2.carbon.utils;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.http.conn.ssl.AbstractVerifier;
+
+import javax.net.ssl.SSLException;
+
+/**
+ * Custom hostname verifier class.
+ */
+public class CustomHostNameVerifier extends AbstractVerifier {
+
+    private final static String[] LOCALHOSTS = {"::1", "127.0.0.1", "localhost", "localhost.localdomain"};
+
+    @Override
+    public void verify(String s, String[] strings, String[] subjectAlts) throws SSLException {
```

Comment:
```suggestion
    public void verify(String hostname, String[] commonNames, String[] subjectAlternativeNames) throws SSLException {
```
what do you think of these variable names?